src/interfaces/database.js

```diff
@@ -4,7 +4,8 @@ export interface Database {
   getBestBlockNum(): any;
   storeUtxos(utxos: Array<mixed>): Promise<any>;
   storeBlockTxs(block: any): Promise<void>;
-
+  genesisLoaded(): Promise<boolean>;
```

Comment:
Maybe this could be isGenesisLoaded()

Reply:
Agree, but let's make it a separate PR for review sake